src/jit/lsra.cpp

```diff
@@ -6709,6 +6809,13 @@ void LinearScan::processBlockStartLocations(BasicBlock* currentBlock, bool alloc
                     }
                     inVarToRegMap[assignedInterval->getVarIndex(compiler)] = REG_STK;
                 }
+#ifdef _TARGET_ARM_
+                // Consider overlapping floating point register for TYP_DOUBLE
+                else if (!assignedInterval->isConstant && isSecondHalfReg(physRegRecord, assignedInterval))
+                {
+                    assert(assignedInterval->assignedReg->regNum == REG_PREV(physRegRecord->regNum));
```

Comment:
This assert basically duplicates what is already checked by `isSecondHalfReg`, but doesn't check for the case where, somehow, `assignedInterval` is a `TYP_DOUBLE` for which the second half has not been appropriately updated. I think a better check would be:
```
else if (!assignedInterval->isConstant && assignedInterval->registerType == TYP_DOUBLE)
{
    assert(!assignedInterval->isActive || isSecondHalfReg(physRegRecord, assignedInterval));
}
```
Does that make sense?